internal/charmstore/server.go

```diff
@@ -46,6 +58,11 @@ func NewServer(db *mgo.Database, si *SearchIndex, config ServerParams, versions
 	if len(versions) == 0 {
 		return nil, errgo.Newf("charm store server must serve at least one version of the API")
 	}
+	config.IdentityLocation = strings.Trim(config.IdentityLocation, "/")
+	config.IdentityAPIURL = strings.Trim(config.IdentityAPIURL, "/")
+	if config.IdentityLocation == "" && config.IdentityAPIURL != "" {
+		config.IdentityLocation = config.IdentityAPIURL + "/v1/discharger"
```

Comment:
Cool fallback, but i wonder if we should also write in a log that the service did this even though the config is different.

Another example is: what if IdentityLocation and IdentityAPI are different. I'd at least write a warning in a log in this case.